### PR TITLE
[risk=low][RW-5564] Default to latest CDR on Workspace dupe and show upgrade message

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

This is only a **partial** completion of RW-5564.  The warning modal for non-latest CDR Version is not implemented here.

This also changes the Workspace Duplication UX to always choose the latest CDR Version for duplication.  It **does not** prevent choosing an older version intentionally.  It also does not warn when doing so - that will be part of the next RW-5564 PR.

dupe-upgrade

Note: "v3 with Microarray" is the older version of the CDR in my local environment.  In production it will likely look more like "you are upgrading from v5 to v8"

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [x] I have run the E2E tests on ths change against my local UI + API server with `yarn test-local`
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: e2e/tests/workspace/workspace-duplicate.spec.ts [hunk 1]
```diff
@@ -76,4 +77,85 @@ describe('Duplicate workspace', () => {
...
+    // Click on Ellipsis menu "Duplicate" option.
+    await workspaceCard.selectSnowmanMenu(Option.Duplicate);
+
+    // Fill out Workspace Name should be just enough for successful duplication
+    const workspacesPage = new WorkspacesPage(page);
+    await (await workspacesPage.getWorkspaceNameTextbox()).clear();
+    const duplicateWorkspaceName = await workspacesPage.fillOutWorkspaceName();
+
+    // change CDR Version
+    await workspacesPage.selectCdrVersion(config.altCdrVersionName);
```
Comment (jmthibault79) [inline e2e/tests/workspace/workspace-duplicate.spec.ts:94]: the next PR will also look for a new "are you sure?" popup here

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 2]
```diff
@@ -315,7 +346,8 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
      * "reviewRequested" flag, which depend on the workspace state & edit mode.
      */
     createInitialWorkspaceState(): Workspace {
-      let workspace: Workspace = this.props.workspace;
+      // copy the props into a new object so our modifications here don't affect them
+      let workspace: Workspace = {...this.props.workspace};
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:350]: Also fixes a bug: Duplication screen showed "Duplicate of Duplicate of My Workspace"

File: ui/src/app/utils/authorities.spec.tsx [hunk 3]
```diff
@@ -10,7 +10,7 @@ const accessAuth = {...ProfileStubVariables.PROFILE_STUB, authorities: [Authorit
 const devAuth = {...ProfileStubVariables.PROFILE_STUB, authorities: [Authority.DEVELOPER]}
 const featuredWsAuth = {...ProfileStubVariables.PROFILE_STUB, authorities: [Authority.FEATUREDWORKSPACEADMIN]}
 
-describe('auth', () => {
+describe('authorities', () => {
```
Comment (jmthibault79) [inline ui/src/app/utils/authorities.spec.tsx:13]: unrelated

File: e2e/tests/workspace/workspace-duplicate.spec.ts [hunk 4]
```diff
@@ -76,4 +77,85 @@ describe('Duplicate workspace', () => {
...
+    // Fill out Workspace Name should be just enough for successful duplication
+    const workspacesPage = new WorkspacesPage(page);
+    await (await workspacesPage.getWorkspaceNameTextbox()).clear();
+    const duplicateWorkspaceName = await workspacesPage.fillOutWorkspaceName();
+
+    // change CDR Version
+    await workspacesPage.selectCdrVersion(config.defaultCdrVersionName);
+
+    const upgradeMessage = await workspacesPage.getCdrVersionUpgradeMessage();
+    expect(upgradeMessage).toContain(`You're duplicating the workspace "${originalWorkspaceName}" to upgrade from`);
```
Comment (petesantos) [inline e2e/tests/workspace/workspace-duplicate.spec.ts:136]: nit: is the full text needed for the test? Would the `originalWorkspaceName` be enough?

File: ui/src/app/pages/workspace/workspace-edit.spec.tsx [hunk 5]
```diff
@@ -216,6 +215,43 @@ describe('WorkspaceEdit', () => {
...
+
+    const wrapper = component();
+    await waitOneTickAndUpdate(wrapper);
+
+    const cdrSelection = wrapper.find('[data-test-id="select-cdr-version"]').find('select').props().value;
+
+    // default CDR version, not the existing workspace's alt CDR version
+    expect(cdrSelection).toBe(CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID);
+
+    const expectedUpgradeMessage1 = `You're duplicating the workspace "${altCdrWorkspace.name}" to upgrade from`;
```
Comment (petesantos) [inline ui/src/app/pages/workspace/workspace-edit.spec.tsx:234]: nit: Same comment as above. The only reason I call it out is to prevent needing to change the test for minor language changes.

**Review by petesantos — APPROVED** [review]
LGTM
Some minor nits
